src/libs/HttpUtils.ts

```diff
@@ -91,43 +89,42 @@ function processHTTPRequest(url, method = 'get', body = null, canCancel = true)
             if (response.jsonCode === CONST.JSON_CODE.EXP_ERROR && response.title === CONST.ERROR_TITLE.SOCKET && response.type === CONST.ERROR_TYPE.SOCKET) {
                 throw new HttpsError({
                     message: CONST.ERROR.EXPENSIFY_SERVICE_INTERRUPTED,
-                    status: CONST.JSON_CODE.EXP_ERROR,
+                    status: CONST.JSON_CODE.EXP_ERROR.toString(),
                     title: CONST.ERROR_TITLE.SOCKET,
                 });
             }
             if (response.jsonCode === CONST.JSON_CODE.MANY_WRITES_ERROR) {
-                const {phpCommandName, authWriteCommands} = response.data;
-                // eslint-disable-next-line max-len
-                const message = `The API call (${phpCommandName}) did more Auth write requests than allowed. Count ${authWriteCommands.length}, commands: ${authWriteCommands.join(
-                    ', ',
-                )}. Check the APIWriteCommands class in Web-Expensify`;
-                alert('Too many auth writes', message);
+                if (response.data) {
+                    const {phpCommandName, authWriteCommands} = response.data;
+                    // eslint-disable-next-line max-len
+                    const message = `The API call (${phpCommandName}) did more Auth write requests than allowed. Count ${authWriteCommands.length}, commands: ${authWriteCommands.join(
+                        ', ',
+                    )}. Check the APIWriteCommands class in Web-Expensify`;
+                    alert('Too many auth writes', message);
+                }
             }
-            return response;
+            return response as Promise<Response>;
         });
 }
 
 /**
  * Makes XHR request
- * @param {String} command the name of the API command
- * @param {Object} data parameters for the API command
- * @param {String} type HTTP request type (get/post)
- * @param {Boolean} shouldUseSecure should we use the secure server
- * @returns {Promise}
+ * @param command the name of the API command
+ * @param data parameters for the API command
+ * @param type HTTP request type (get/post)
+ * @param shouldUseSecure should we use the secure server
  */
-function xhr(command, data, type = CONST.NETWORK.METHOD.POST, shouldUseSecure = false) {
+function xhr(command: string, data: Record<string, unknown>, type: RequestType = CONST.NETWORK.METHOD.POST, shouldUseSecure = false): Promise<Response> {
     const formData = new FormData();
-    _.each(data, (val, key) => {
-        // Do not send undefined request parameters to our API. They will be processed as strings of 'undefined'.
-        if (_.isUndefined(val)) {
+    Object.keys(data).forEach((key) => {
+        if (!data[key]) {
```

Comment:
This is not equivalent to the old code... this will remove any `0`, `false, `null` or `""` while the old code would not

Reply:
@jjcoffee FYI since you missed this in your review and I have the feeling this would've broken a ton of things.

Reply:
Damn, good catch! Sorry for missing that :bow: